### PR TITLE
[ADD] consider-merging-isinstance: Add new check consider-merging-isinstance

### PR DESCRIPTION
**Pull request by moylop260**
### Fixes / new features
- Fix https://github.com/PyCQA/pylint/issues/968

- [x] Consider just `or` cases (Discard `not`, `and`, ...)
- [x] Move extension to core
- [x] Add doc

- [x] Fix the following cases not detected: 
 - [pylint/checkers/base.py#L186-L187](https://github.com/PyCQA/pylint/blob/5d40a07684be7ed6dbaa54e49c3d436005a533bf/pylint/checkers/base.py#L186-L187)
 - [pylint/checkers/variables.py#L464-L465](https://github.com/PyCQA/pylint/blob/5d40a07684be7ed6dbaa54e49c3d436005a533bf/pylint/checkers/variables.py#L464-L465)
 - The above examples are good not-grouped cases, because they are using: `isinstance(obj1, class_x) and obj1.value1 or isinstance(obj1, class_y) and obj1.value2` and this is a valid not-grouped.

- [x] Maybe we should try to print how should it look exactly?
So in the case of if isinstance(var[3], int) or isinstance(var[3], float) or isinstance(var[3], list) it can look as:
$ pylint a.py
1:R:Consider merging these calls to:
     isinstance(var[3], (int, float, list))

### CODE REVIEW

File: pylint/extensions/check_ungrouped_isinstances.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+"""Module to check ungrouped isinstances methods for pylint. """
+
+import astroid
+
+from pylint import checkers, utils
+from pylint.checkers.utils import check_messages
+from pylint.interfaces import IAstroidChecker
+
+MSGS = {
+    'E2685': ("isinstances of '%s' are not grouped. "
```
Comment (PCManticore) [inline pylint/extensions/check_ungrouped_isinstances.py]: Has to be a Refactoring message, not an Error.

Comment (PCManticore) [inline pylint/extensions/check_ungrouped_isinstances.py]: Please rename it to consider-merging-isinstance and change the title and description of the message accordingly.

File: pylint/extensions/check_ungrouped_isinstances.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    def visit_boolop(self, node):
+        "Check ungrouped isinstances"
+        if node.op != 'or':
+            return
+        first_args = [
+            value.args[0].as_string()
+            for value in node.values
+            if isinstance(value, astroid.Call) and value.args and
+            value.func.name == 'isinstance']
+        for duplicated_name in utils.get_duplicated(first_args):
```
Comment (PCManticore) [inline pylint/extensions/check_ungrouped_isinstances.py]: You have to check it is also a builtin. Also, let the inference do its job here, infer what value.func is and then check the name, only if it is a builtin.

File: pylint/utils.py [hunk 3]
```diff
@@ -1145,3 +1145,19 @@ def _rest_format_section(stream, section, options, encoding=None, doc=None):
...
+
+
+def get_duplicated(items):
+    """Get duplicated items
+    :param iter items: Items to extract duplicated
+    :returns: Duplicated extracted from items
+    :rtype: list
+    """
+    unique_items = set()
+    duplicated_items = set()
```
Comment (PCManticore) [inline pylint/utils.py]: You can use a collections.Counter here. This might look as (untested):

c = collections.Counter(items)
return [key for key, value in c.items() if value == 1]

Comment (PCManticore) [inline pylint/utils.py]: Typo, > of course

File: pylint/test/extensions/test_check_ungrouped_isinstances.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+"""Tests for the pylint checker in :mod:`pylint.extensions.check_ungrouped_isinstances
+"""
+
+import os.path as osp
+import unittest
+
+from pylint import checkers
+from pylint.extensions.check_ungrouped_isinstances import register
```
Comment (PCManticore) [inline pylint/test/extensions/test_check_ungrouped_isinstances.py]: I am considering moving this under the refactoring checker, so this test would be gone and we could use proper functional tests.

**Comment by moylop260** [comment]
> @PCManticore 
I am considering moving this under the refactoring checker, so this test would be gone and we could use proper functional tests.

I moved the test to `functional` and I verify that work the rc file with the following command:
`python -m pylint --rcfile=pylint/test/functional/consider_merging_isinstance.rc pylint/test/functional/consider_merging_isinstance.py`

The result was:
```bash
Problem importing module variables.pyc: No module named functools_lru_cache
Problem importing module variables.py: No module named functools_lru_cache
************* Module functional.consider_merging_isinstance
R: 14, 7: isinstances of 'var3' are not merged. Consider merging 'isinstance(var3, (class1, class2, ...))' (consider-merging-isinstance)
R: 16,13: isinstances of 'var4' are not merged. Consider merging 'isinstance(var4, (class1, class2, ...))' (consider-merging-isinstance)
R: 18,13: isinstances of 'var5' are not merged. Consider merging 'isinstance(var5, (class1, class2, ...))' (consider-merging-isinstance)
R: 21,13: isinstances of 'var6' are not merged. Consider merging 'isinstance(var6, (class1, class2, ...))' (consider-merging-isinstance)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)
```

But running `tox -e py27` I get the following error:
```bash
py27 runtests: PYTHONHASHSEED='4194164116'                                                                                                            [21/1775]
py27 runtests: commands[0] | python -Wi /root/build/vauxoo-dev/pylint/.tox/py27/lib/python2.7/site-packages/coverage run -m unittest discover -s /root/build/va
uxoo-dev/pylint/.tox/py27/lib/python2.7/site-packages/pylint/test/ -p *test_*.py
..............................s................................................................................................................................
...........................s...................................................................................................................................
...........................................sss................sss..................................................s...x.......s.....s.s....s..sss.......s.s...
.s.......s...............s..s......s.s...............s.s......s.s...s.......s...s........s..................s..s.....s....s...s..ss...s.....sss..s......s.F....
..s......s......s...s.s....s....s...
======================================================================
FAIL: consider_merging_isinstance (test_functional.LintModuleTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/build/vauxoo-dev/pylint/.tox/py27/lib/python2.7/site-packages/pylint/test/test_functional.py", line 311, in _runTest
    self.fail('\n'.join(msg))
AssertionError: Wrong results for file "consider_merging_isinstance":

Expected in testdata:
  14: consider-merging-isinstance
  16: consider-merging-isinstance
  18: consider-merging-isinstance
  21: consider-merging-isinstance

----------------------------------------------------------------------
Ran 672 tests in 26.162s

FAILED (failures=1, skipped=52, expected failures=1)
```

Could you help to give a tip to enable a `extension` from `functional test`?
Or maybe I don't understood your last comment

**Comment by PCManticore** [comment]
I am not sure why do you get ``Problem importing module variables.pyc: No module named functools_lru_cache``, maybe you need to do another clean install to get all the new dependencies?

Regarding my comment, I was saying that this could be a new core checker instead of an extension. It looks reasonable and might be okay to emit a refactoring message for it. Given this fact, you just have to modify the rcfile to not load the extension. For some reason, we can't load the extensions in those particular tests, needs some investigation, but it is pretty low priority.

**Comment by moylop260** [comment]
Thanks for replying...

Should I fix it with the following change https://github.com/PyCQA/pylint/pull/976/commits/2f047a948a8d27b1ac52d9f1e928ba9b359b961f?
Or Should I move the extension check to a normal check?

**Comment by moylop260** [comment]
What is the better place to move this check from extension to core `pylint/checkers/stdlib.py` maybe?

**Comment by PCManticore** [comment]
I would say design_analysis.py. Or we can make a new checker, refactoring.py and we can move there the checks which are dealing with refactoring per se, such as considering-using-enumerate and company.

File: pylint/extensions/check_consider_merging_isinstance.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    def visit_boolop(self, node):
+        "Check not merged isinstance"
+        if node.op != 'or':
+            return
+        first_args = [
+            value.args[0].as_string()
+            for value in node.values
+            if isinstance(value, astroid.Call) and value.args and
+            is_builtin_object(safe_infer(value.func)) and
+            safe_infer(value.func).name == 'isinstance']
```
Comment (PCManticore) [inline pylint/extensions/check_consider_merging_isinstance.py]: This can crash, since safe_infer can return None. Also, you can refactor this list comprehension into a function generator, which yields the valid arguments out of it.

File: pylint/utils.py [hunk 6]
```diff
@@ -1145,3 +1145,13 @@ def _rest_format_section(stream, section, options, encoding=None, doc=None):
             value = _encode(_format_option_value(optdict, value), encoding)
             print(file=stream)
             print('  Default: ``%s``' % value.replace("`` ", "```` ``"), file=stream)
+
+
+def get_duplicated(items):
```
Comment (PCManticore) [inline pylint/utils.py]: This can be local in the file where the check is, since it is not yet used by more than one checker. Also, it could be renamed to duplicated(items).

**Comment by PCManticore** [comment]
Can you also add a ChangeLog entry, plus a new entry into *What's New* section ( https://pylint.readthedocs.io/en/latest/whatsnew/2.0.html)? You can create a new category, "Added features" or something and it can have a more detailed description of what the check is actually doing. Right now that document is quite empty, but I plan to write about the 2.0 changes there.

File: pylint/test/test_functional.py [hunk 7]
```diff
@@ -226,6 +230,7 @@ def __init__(self, test_file):
         self._linter.set_reporter(test_reporter)
         self._linter.config.persistent = 0
         checkers.initialize(self._linter)
+        register_check_consider_merging_isinstance(self._linter)
```
Comment (PCManticore) [inline pylint/test/test_functional.py]: You don't have to do this here. Even if we are moving the checker into the core, you should not change test_functional in this way for registering additional checkers.

File: pylint/utils.py [hunk 8]
```diff
@@ -1145,3 +1145,13 @@ def _rest_format_section(stream, section, options, encoding=None, doc=None):
...
+    """Get duplicated items
+    :param iter items: Items to extract duplicated
+    :returns: Duplicated extracted from items
```
Comment (PCManticore) [inline pylint/utils.py]: "Duplicated items retrieved from the given argument."

File: doc/whatsnew/2.0.rst [hunk 9]
```diff
@@ -49,3 +49,11 @@ Removed Changes
...
   code was parsed by Python as well, without actually reaching the
   runtime step and as such, we decided to remove the error altogether.
+
+
+Added features
+==============
+
+* New refactory message 'consider-merging-isinstance', which is emitted
+  when the built-in method `isinstance` is used many times with
+  the same object in `or` operations.
```
Comment (PCManticore) [inline doc/whatsnew/2.0.rst]: This would need an example of what is correct and what is not.

File: pylint/checkers/refactoring.py [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+MSGS = {
+    'R1701': (
+        "isinstances of '%s' are not merged. "
+        "Consider merging 'isinstance(%s, (class1, class2, ...))'",
+        "consider-merging-isinstance",
+        "Used when isinstance methods are not grouped by object"),
+}
+
+
+def get_duplicated(items):
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: ``def _duplicated`` instead.

File: pylint/checkers/utils.py [hunk 11]
```diff
@@ -244,8 +244,8 @@ def is_func_decorator(node):
         if isinstance(parent, astroid.Decorators):
             return True
         if (parent.is_statement or
-                isinstance(parent, astroid.Lambda) or
-                isinstance(parent, (scoped_nodes.ComprehensionScope,
+                isinstance(parent, (astroid.Lambda,
```
Comment (PCManticore) [inline pylint/checkers/utils.py]: Why did you change this?

File: ChangeLog [hunk 12]
```diff
@@ -84,6 +84,10 @@ Release date: tba
       emitted when item deletion is tried on an object which doesn't
       have this ability. Closes issue #592.
 
+    * Added a new refactory message, 'consider-merging-isinstance', which is
```
Comment (PCManticore) [inline ChangeLog]: `refactoring message``

File: ChangeLog [hunk 13]
```diff
@@ -84,6 +84,10 @@ Release date: tba
...
+      emitted when the built-in method `isinstance` is used many times with
```
Comment (PCManticore) [inline ChangeLog]: ``is used to check the type of the same object, with different calls, in the same chained boolean operation``.

File: doc/whatsnew/2.0.rst (hunk 9 above)
Comment (PCManticore) [inline doc/whatsnew/2.0.rst]: Also, modify it accordingly after changing the entry from the Changelog.

File: pylint/checkers/refactoring.py [hunk 14]
```diff
@@ -0,0 +1,77 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+"""check to consider a refactory of code"""
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: "Looks for code which can be refactored."

File: pylint/checkers/refactoring.py [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+import astroid
+
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages, is_builtin_object, safe_infer
+
+
+MSGS = {
+    'R1701': (
+        "isinstances of '%s' are not merged. "
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: ``Consider merging these isinstance calls to %s.``

File: pylint/checkers/refactoring.py [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+import collections
+
+import astroid
+
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages, is_builtin_object, safe_infer
+
+
+MSGS = {
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: This can be part of the checker itself, instead of being global

File: pylint/checkers/refactoring.py [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+    'R1701': (
+        "isinstances of '%s' are not merged. "
+        "Consider merging 'isinstance(%s, (class1, class2, ...))'",
+        "consider-merging-isinstance",
+        "Used when isinstance methods are not grouped by object"),
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: ``Usen when multiple isinstance calls can be grouped into one.``

File: pylint/checkers/refactoring.py (hunk 10 above)
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: If this is private, you can drop the documentation. The name is enough for determining what it does.

File: pylint/checkers/refactoring.py [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+    """Get duplicated items
+    :param iter items: Items to extract duplicated
+    :returns: Duplicated items retrieved from the given argument.
+    :rtype: list
+    """
+    items_counter = collections.Counter(items)
+    return [item for item, counter in items_counter.items() if counter > 1]
+
+
+class RefactoryChecker(BaseChecker):
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: RefactoringChecker.

File: pylint/checkers/refactoring.py [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+    """Checks to consider a refactory of code
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: ``Looks for code which can be refactored.`` is enough.

File: pylint/checkers/refactoring.py [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+    * isinstance merging
+    """
+
+    __implements__ = (IAstroidChecker,)
+
+    # configuration section name
+    name = 'refactory'
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: ``refactoring``

File: pylint/checkers/refactoring.py [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+
+    msgs = MSGS
+    priority = -2
+
+    @staticmethod
+    def _get_first_args(node):
+        # pylint: disable=redundant-returns-doc
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: Please create an issue if we have these problems.

File: pylint/checkers/refactoring.py [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+        # yield is a type of return
+        """Get first item of all node.args as string of method 'isinstance'.
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: ``Get the objects, as strings, from the *isinstance* calls, found in the BoolOp node.``

File: pylint/checkers/refactoring.py [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+                continue
+            inferred = safe_infer(value.func)
+            if not inferred or not is_builtin_object(inferred):
+                continue
+            if inferred.name == 'isinstance':
+                yield value.args[0].as_string()
+
+    @check_messages(*(MSGS.keys()))
+    def visit_boolop(self, node):
+        "Check not merged isinstance"
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: ``Check isinstance calls which can be merged together.``

File: pylint/checkers/refactoring.py [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+        for value in node.values:
+            if not isinstance(value, astroid.Call) or not value.args:
+                continue
+            inferred = safe_infer(value.func)
+            if not inferred or not is_builtin_object(inferred):
+                continue
+            if inferred.name == 'isinstance':
+                yield value.args[0].as_string()
+
+    @check_messages(*(MSGS.keys()))
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: Please enable only what it is expected by this visit method to check, for instance, in this case is only ``consider-merging-isinstances``. Otherwise, this would be a source of bugs when we will add more checks here.

File: pylint/test/functional/consider_merging_isinstance.py [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+
+    # not merged
+    if isinstance(var3, int) or isinstance(var3, long) or isinstance(var3, list) and True:  # [consider-merging-isinstance]
+        pass
+    result = isinstance(var4, int) or isinstance(var4, long) or isinstance(var5, list) and False  # [consider-merging-isinstance]
+    result = isinstance(var7, int) or not isinstance(var7, long)
+    result = isinstance(var5, int) or True or isinstance(var5, long)  # [consider-merging-isinstance]
+
+    infered_isinstance = isinstance
+    result = infered_isinstance(var6, int) or infered_isinstance(var6, long) or infered_isinstance(var6, list) and False   # [consider-merging-isinstance]
```
Comment (PCManticore) [inline pylint/test/functional/consider_merging_isinstance.py]: Add a test for different objects, such as ``isinstance(x, int) or insinstance(z, int)``

File: pylint/test/functional/consider_merging_isinstance.rc [hunk 26]
```diff
@@ -0,0 +1,3 @@
+[Messages Control]
```
Comment (PCManticore) [inline pylint/test/functional/consider_merging_isinstance.rc]: You don't need this file now.

File: pylint/checkers/utils.py (hunk 11 above)
Comment (moylop260) [inline pylint/checkers/utils.py]: The build `TOXENV=pylint` show me `consider-merging-isinstance` here.
Should I revert it and use a `pylint:disable=consider-merging-isinstance`?

Comment (PCManticore) [inline pylint/checkers/utils.py]: Oh, okay, I didn't notice, I thought you were changing the indentation.

File: pylint/checkers/refactoring.py (hunk 21 above)
Comment (moylop260) [inline pylint/checkers/refactoring.py]: https://github.com/PyCQA/pylint/issues/984

**Comment by PCManticore** [comment]
Disabling coveralls comments after all, they are seriously annoying. The problem is probably the fact that we do not combine the reports from all the builds, which is why we have coverage decreasing and increasing all the time.

File: ChangeLog [hunk 27]
```diff
@@ -84,6 +84,11 @@ Release date: tba
       emitted when item deletion is tried on an object which doesn't
       have this ability. Closes issue #592.
 
+    * Added a new refactoring message, 'consider-merging-isinstance', which is
+      used to check the type of the same object, with different calls, in the
```
Comment (PCManticore) [inline ChangeLog]: This is a bit wrong, since it says how the check is implemented, but not what it does.

Maybe ``Added a new refactoring message, ..., which is used to detect isinstance calls, which can be merged together.`` would be better.

File: doc/whatsnew/2.0.rst [hunk 28]
```diff
@@ -49,3 +49,13 @@ Removed Changes
...
   runtime step and as such, we decided to remove the error altogether.
+
+
+Added features
+==============
+
+* Refactoring message, 'consider-merging-isinstance', which is
+  used to check the type of the same object, with different calls, in the
+  same chained boolean operation.
+  Consider merging instance `isinstance(var1, (int, long))` instead of
```
Comment (PCManticore) [inline doc/whatsnew/2.0.rst]: This needs an actual example and please remove the last part. So something as:

    .. code-block:: python

       $ cat a.py
       isinstance(x, int) or isinstance(z, int)
       $ pylint a.py -rn
       # pylint's output here

Also, modify the description to say a similar thing as the one from the ChangeLog. Right now I think it is a bit confusing.

File: pylint/checkers/refactoring.py [hunk 29]
```diff
@@ -0,0 +1,68 @@
...
+import collections
+
+import astroid
+
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages, is_builtin_object, safe_infer
+
+
+def _get_duplicated(items):
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: Drop the _get.

File: pylint/checkers/refactoring.py [hunk 30]
```diff
@@ -0,0 +1,68 @@
...
+    msgs = {
+        'R1701': (
+            "Consider merging these isinstance calls to %s.",
+            "consider-merging-isinstance",
+            "Usen when multiple isinstance calls can be grouped into one."),
+    }
+    priority = -2
+
+    @staticmethod
+    def _get_first_args(node):
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: Drop the _get

File: pylint/test/functional/consider_merging_isinstance.txt [hunk 31]
```diff
@@ -0,0 +1,5 @@
+consider-merging-isinstance:15:isinstances:Consider merging these isinstance calls to var[3].
```
Comment (PCManticore) [inline pylint/test/functional/consider_merging_isinstance.txt]: Maybe we should try to print how should it look exactly?
So in the case of `` if isinstance(var[3], int) or isinstance(var[3], float) or isinstance(var[3], list)`` it can look as:

    $ pylint a.py
    1:R:Consider merging these calls to:
         isinstance(var[3], (int, float, list))

Comment (moylop260) [inline pylint/test/functional/consider_merging_isinstance.txt]: This is a good approach.
This require a full refactoring of the current methods.
Are you agree if I create a new issue with this request and I work it in other PR?

File: doc/whatsnew/2.0.rst [hunk 32]
```diff
@@ -49,3 +49,16 @@ Removed Changes
...
+==============
+
+* Refactoring message, 'consider-merging-isinstance', which is
+  used to detect isinstance calls, which can be merged together. 
+
+.. code-block:: python
+
+    $ cat a.py
+    isinstance(x, int) or isinstance(x, float)
+    # R:  1, 0: Consider merging these isinstance calls to x. (consider-merging-isinstance)
```
Comment (PCManticore) [inline doc/whatsnew/2.0.rst]: Forgot a:

    $ pylint a.py
    # 1, 0 ....

File: pylint/checkers/refactoring.py [hunk 33]
```diff
@@ -0,0 +1,68 @@
...
+            "Consider merging these isinstance calls to %s.",
+            "consider-merging-isinstance",
+            "Usen when multiple isinstance calls can be grouped into one."),
+    }
+    priority = -2
+
+    @staticmethod
+    def first_args(node):
+        # TODO: Remove after fix https://github.com/PyCQA/pylint/issues/984
+        # pylint: disable=redundant-returns-doc
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: Should be fixed now.

File: pylint/checkers/refactoring.py [hunk 34]
```diff
@@ -0,0 +1,68 @@
...
+    msgs = {
+        'R1701': (
+            "Consider merging these isinstance calls to %s.",
+            "consider-merging-isinstance",
+            "Usen when multiple isinstance calls can be grouped into one."),
+    }
+    priority = -2
+
+    @staticmethod
+    def first_args(node):
```
Comment (PCManticore) [inline pylint/checkers/refactoring.py]: Make it private.

File: pylint/test/functional/consider_merging_isinstance.txt (hunk 31 above)
Comment (PCManticore) [inline pylint/test/functional/consider_merging_isinstance.txt]: Why would it require a full refactoring? It could be done in this PR, we will merge it after it.

Comment (moylop260) [inline pylint/test/functional/consider_merging_isinstance.txt]: Done

**Comment by PCManticore** [comment]
Merged locally: https://github.com/PyCQA/pylint/commit/5c40fd9ae3c7c8e8181a473d7708ed670924cc6d. Also I cleaned it up a bit in https://github.com/PyCQA/pylint/commit/7303fd6948f38867c32b32f9584d1fd768b3b6d6
